compiler-lib/src/ast.rs

```diff
@@ -201,3 +201,30 @@ pub enum UnaryOp {
 }
 
 pub type ArgumentList<'t> = Vec<Spanned<'t, Expr<'t>>>;
+
+impl<'f> ClassMemberKind<'f> {
+    pub fn is_method(&self) -> bool {
+        use self::ClassMemberKind::*;
+        match self {
+            Method(_, _, _) => true,
+            MainMethod(_, _) => true,
+            Field(_) => false,
+        }
+    }
+    pub fn method_params(&self) -> Option<&Spanned<'_, ParameterList<'_>>> {
+        use self::ClassMemberKind::*;
+        match &self {
+            Method(_t, pl, _block) => Some(pl),
+            MainMethod(pl, _block) => Some(pl),
```

Comment:
```suggestion
            MainMethod(_, _, _) => None,
```

Reply:
due to type inference, the variable name is the only way to catch changed semantics (e.g. different ordering of struct members) during code review. but then again, the order-changing should have not passed code review anyways...